src/Symfony/Component/Finder/Iterator/ExcludeDirectoryFilterIterator.php

```diff
@@ -42,14 +44,23 @@ public function __construct(\Iterator $iterator, array $directories)
      */
     public function accept()
     {
-        $path = $this->isDir() ? $this->current()->getRelativePathname() : $this->current()->getRelativePath();
+        $path = $this->current();
+
+        if (isset($this->patterns[$path->getFilename()])) {
+            return $this->acceptChildren = false;
+        }
+        if ($this->isRecursive) {
```

Comment:
this is broken in case you the excluded directory is ``tests/cache`` rather than a short class name (which is supported, and even the only way to exclude ``tests/cache`` without excluding ``other/cache`` too)